tokio/src/runtime/spawner.rs

```diff
@@ -25,15 +25,19 @@ impl Spawner {
         }
     }
 
-    pub(crate) fn spawn<F>(&self, future: F, id: Id) -> JoinHandle<F::Output>
+    pub(crate) fn spawn<F>(
+        &self,
+        future: F,
+        id: Id,
+    ) -> Result<JoinHandle<F::Output>, SpawnFailure<F::Output, SpawnError>>
     where
         F: Future + Send + 'static,
         F::Output: Send + 'static,
     {
         match self {
             Spawner::CurrentThread(spawner) => spawner.spawn(future, id),
             #[cfg(all(feature = "rt-multi-thread", not(tokio_wasi)))]
-            Spawner::MultiThread(spawner) => spawner.spawn(future, id),
+            Spawner::MultiThread(spawner) => Ok(spawner.spawn(future, id)),
         }
```

Comment:
This always returns `Ok` in the `MultiThread` case, but the spawn would fail if the runtime is shut down.

Reply:
Do you know where the rejection happens? I tried to trace the spawner internals but eventually I gave up as it seemed infallible

Happy to bubble it up appropriately though

Reply:
A found it in `bind_new_task`, will update